### PR TITLE
TypeChecking on LIBs active

### PR DESCRIPTION
**Pull request by jacksonic**
Memoize0/1 retains the original function's toString() in order to support type checking on memo'd functions.

### CODE REVIEW

File: src/foam/core/debug.js [hunk 1]
```diff
@@ -368,44 +393,78 @@ foam.LIB({
         }
 
         // If nothing threw an exception, we are free to run the function
-        var retVal = fn.apply(this, arguments);
+        var typeCheckerVal = fn.apply(this, arguments);
 
-        // check the return value
-        if ( args.returnType ) args.returnType.validate(retVal);
+        // check the typeCheckerurn value
```
Comment (kgrgreer) [inline src/foam/core/debug.js]: What is 'typeCheckerurn'?

File: src/foam/core/stdlib.js [hunk 2]
```diff
@@ -381,12 +371,9 @@ foam.LIB({
 
     {
       name: 'camelize',
```
Comment (kgrgreer) [inline src/foam/core/stdlib.js]: Move out of stdlib.

Comment (jacksonic) [reply src/foam/core/stdlib.js]: Done on master

File: src/foam/core/debug.js (hunk 1 above)
Comment (jacksonic) [reply src/foam/core/debug.js]: fixed... search and replace typp

**Pull request merged**